.github/workflows/build_for_pypi.yml

```
@@ -1,46 +1,90 @@
+---
 name: Build and Optionally Publish to PyPi
 
 on:
-  workflow_call:
-    inputs:
-      publish:
-        type: boolean
-        default: false
-        description: "Publish to PyPi"
+  pull_request:
+    branches:
+      - main
+  push:
+    # Pattern matched against refs/tags
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
 
 jobs:
-  build_for_pypi:
+  build_dist_for_pypi:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-13, macos-latest]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          submodules: true
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.22.0
```

This should be able to pick up that sdist and build from it instead of building from Git:
```suggestion
        uses: pypa/cibuildwheel@v2.22.0
        with:
          package-dir: dists/codecov*.tar.gz
```
(yes, the input name is a bit unobvious, I already reported it upstream in the past: https://github.com/pypa/cibuildwheel/issues/1682)